src/signature/mechanism/cosign/mod.rs

```diff
@@ -173,8 +177,18 @@ impl CosignParameters {
             }) => Err(anyhow!("{:?}", unsatisfied_constraints)),
         }
     }
+
+    #[cfg(target_arch = "s390x")]
+    async fn verify_signature_and_get_payload(
+        &self,
+        _image: &mut Image,
+        _auth: &RegistryAuth,
+    ) -> Result<Vec<SigPayload>> {
+        Err(anyhow!("Cosign is not supported on IBM s390x"))
```

Comment:
minor wording: _"Cosign is not supported on the s390x architecture"_

Reply:
I changed the wording a bit more because following Aaron's feedback cosign is a feature, so you can built image-rs without support for cosign on, for example, the amd64 architecture if you build image-rs with non-default feature flags which exclude the cosign feature.

Reply:
The new approach looks much cleaner for me! Thanks.